thunder/tests/distributed/test_ddp.py

```diff
@@ -1453,6 +1468,139 @@ def _test_ddp_transformer_engine_llama_sanity(input_data):
     return None
 
 
+def _test_fsdp_transformer_engine(input_data):
+    # Test Description: We run a dummy training loop for a simple `Linear(Relu(Linear(x)))`
+    # model with thunder (using TE executor) and with PyTorch eager + TE
+    # and verify that the weights have converged to same value and
```

Comment:
Does TransformerEngine generally guarantee this? Could this become a source of test flakiness in the future if they diverge beyond certain tolerance?

Reply:
Since, we are using the exact same TE layers under the hood, this should be true within floating point errors, otherwise this will signals that there is a bug in the implementation.